auth/src/main/java/com/firebase/ui/auth/provider/TwitterProvider.java

```diff
@@ -60,8 +74,36 @@ public void startLogin(Activity activity) {
     }
 
     @Override
-    public void success(Result<TwitterSession> result) {
-        mCallbackObject.onSuccess(createIdpResponse(result.data));
+    public void success(final Result<TwitterSession> sessionResult) {
```

Comment:
@SUPERCILEX I appreciate all the investigation but I have a few reservations about this approach:

- We are supposed to be implementing a best-practices UI layer on top of various auth SDKs, but this is clearly not what Twitter recommends as best practice.
- This adds a lot of complexity by introducing Retrofit into our code (I know it was a dependency before, but now we are using it directly)
- The benefit here is real, but it's not huge.  I think I'd rather use the Twitter SDK as it stands and just deal with the ambiguity of their errors.

As for the leak, I think we can work around that ourselves by making the callback a `static` inner class with a `WeakReference` to the `Activity`.